src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -324,6 +324,20 @@ def getTask(self, taskName):
             return None
         return WMTaskHelper(task)
 
+    def getTaskByName(self, taskName):
+        """
+        _getTaskByName_
+
+        Retrieve a task with the given name in the whole workflow tree.
+        """
+        for t in self.taskIterator():
+            if t.name() == taskName:
+                return t
+            for x in t.taskIterator():
```

Comment:
Seangchan, I've just noticed it has to `childTaskIterator` instead of `taskIterator`, otherwise the `tasklist` attribute always yield the same task name.